presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpiller.java

```diff
@@ -76,13 +78,17 @@ public FileSingleStreamSpiller(
             Path spillPath,
             SpillerStats spillerStats,
             SpillContext spillContext,
-            LocalMemoryContext memoryContext)
+            LocalMemoryContext memoryContext,
+            Optional<SpillCipher> spillCipher)
     {
         this.serde = requireNonNull(serde, "serde is null");
         this.executor = requireNonNull(executor, "executor is null");
         this.spillerStats = requireNonNull(spillerStats, "spillerStats is null");
         this.localSpillContext = spillContext.newLocalSpillContext();
         this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        this.spillCipher = requireNonNull(spillCipher, "spillCipher is null");
+        checkState(!spillCipher.isPresent() || !spillCipher.get().isDestroyed(), "spillCipher is already destroyed");
+        this.spillCipher.ifPresent(cipher -> closer.register(cipher::destroy));
```

Comment:
`spillCipher` field is otherwise unused

- no need for this to be a field
- add a comment why you need it at all
  ```
  if (requireNonNull(spillCipher, "spillCipher is null").isPresent()) {
      // spillCipher is not used here. It's passed here only to ensure it's properly destroyed as PagesSerde is not closeable.
      closer.register(spillCipher.get()::destroy);
  }
  ```

Reply:
You are correct, that was accidentally kept around from previous iterations. Will remove the field reference.